Add SSO integration setup (#20199)
docker/docs: content/security/for-admins/single-sign-on/configure/configure-idp.md

{{< tabs >}}
{{< tab name="Okta" >}}

You can configure Docker with Okta using the pre-configured Docker integration, or by creating a custom application. You can use the [Docker app integration](#add-the-app-integration-in-okta) to simplify the set-up process.

### Add the app integration in Okta

1. Sign in to the Okta Console.
2. Select **Applications > Add Application**.
3. Search for Docker.
4. Select **Add**. This opens the **General Settings** page.
5. Configure the following settings:
   - **Application label**: Leave this as Docker or customize if you prefer.
   - **Docker SSO Connection ID**: Copy the **Connection ID** value from Docker Hub or Admin Console and paste it here.
   - **Application visibility**: Optionally, configure if you want this app visible to your users.
6. Select **Done**.

> 7. After you create the app, go to your app and select **View SAML setup instructions**.

Author: Question: Is the **Sign On** tab in the app? So would the steps be Go to your app, then **Sign On** tab, then **More details**?

> 7. After you create the app, go to your app and select **View SAML setup instructions**.

Reviewer: yes is in the app, i shared the screenshot on a comment below

10. Here you can find the **SAML Sign-in URL** and the **x509 Certificate**. Open the certificate file in a text editor and paste the contents of the file in the **x509 Certificate** field in Docker Hub or Admin Console. Then, copy the value of the **SAML Sign-in URL** and paste it into the corresponding field in Docker Hub or Admin Console.

{{< /tab >}}
{{< tab name="Entra ID SAML" >}}

You can configure Docker with Entra ID using the pre-configured Docker integration, or by creating a custom application. You can use the [Docker app integration](#add-the-app-integration-in-entra-id) to simplify the set-up process.

### Add the app integration in Entra ID

1. Go to the Entra ID admin portal.

> 2. Go to **Identity > Applications > Enterprise applications > All applications**.
> 3. Select **New application**.

Reviewer: suggestion: I think this could be reduced to:
```suggestion
2. Go to **Manage > Enterprise applications > New application**.
```

> 4. Search for Docker and add it to the main directory.

Reviewer: suggestion: hightlight "Docker" as search term 
```suggestion
4. Search for **Docker** and select the correct app to **Create** into the main directory.
```

Reviewer: issue: After adding the app to their directory, user needs to navigate into **Setup single sign on > Get started > SAML**

Author: Updating!

5. Copy the Docker SSO **Connection ID** from Docker Hub or Admin Console.

> 6. Enter the SSO Connection ID in Entra ID to create your application.

Reviewer: issue: For the Entra ID app, users need to copy both Entity ID and ACS URL

Author: Question: do they need the Connection ID also or just Entity ID and ACS URL?

Reviewer: just Entity ID and ACS URL, we could use the same screenshots that Microsoft added to their guide, or we can also reference to Microsoft's guide just for additional reference. https://learn.microsoft.com/en-us/entra/identity/saas-apps/docker-tutorial

Author: Thanks, I'll link to their guide 👍

> 7. Once your application is created, find the **Login URL** and **SAML Signing Certificate** and copy them into the corresponding fields in your Docker SSO connection.

Reviewer: suggestion: Specify Certificate base 64 so they don't have an issue on copy pasting the value into Docker Hub. Also change the order as they appear on Entra app
```suggestion
7. Once your application is created, find the **SAML Signing Certificate (Base 64)** and **Login URL** and copy them into the corresponding fields in your Docker SSO connection.
```
